plugins/default/lib/component_health.py

```diff
@@ -79,22 +75,19 @@ def report(summary, test_results):
         print 'Test: %s - Status: %s - Details: %s' % (test['description'], test['test_status'], test['result'])
 
 
-def main():
-    try:
-        args = generate_parser().parse_args()
-        host = args.host
-        port = args.port
-        endpoint = args.endpoint
-        response = do_request(host, port, endpoint)
-        nagios_status, test_summary, test_results = parse_response(response)
-        report(test_summary, test_results)
-    except RequestError as error:
-        print error.message
-        nagios_status = nagios.CRIT
-    except:
-        print "Unexpected error: %s" % (sys.exc_info()[0])
-        nagios_status = nagios.UNKNOWN
-    sys.exit(nagios_status)
+def check(host, port, endpoint):
+    response = do_request(host, port, endpoint)
+    nagios_status, test_summary, test_results = parse_response(response)
+    report(test_summary, test_results)
+    return nagios_status
+
 
 if __name__ == "__main__":
-    main()
+    args = generate_parser().parse_args()
+    code = nagios.UNKNOWN
+    try:
+        code = check(args.host, args.port, args.endpoint)
+    except:
+        traceback.print_exc()
+    finally:
+        sys.exit(code)
```

Comment:
This is now showing a stacktrace and a not very useful message in an error case for a status in the UI of nagios.

```
Status Information:	(No output on stdout) stderr: Traceback (most recent call last):
File "/opt/rhmap/nagios/plugins/fh-check-component-health", line 89, in <module>
code = check(args.host, args.port, args.endpoint)
File "/opt/rhmap/nagios/plugins/fh-check-component-health", line 79, in check
response = do_request(host, port, endpoint)
File "/opt/rhmap/nagios/plugins/fh-check-component-health", line 55, in do_request
(url, error))
RequestError
```

Previously it used to just be:

```
connect to address fh-appstore and port 8080: No route to host
```